### PR TITLE
Add SPI support to STM32L0 and enable SPI1 port on NUCLEO-L073RZ

### PR DESCRIPTION
**Pull request by tagunil**
Tested with spi_loopback test.

### CODE REVIEW

File: ext/hal/st/stm32cube/stm32l0xx/README [hunk 1]
```diff
@@ -1,6 +1,48 @@
...
+   *Current implementation of LL_SPI_TransmitData16 on L0 family
+    generates following warning:
+    "warning: dereferencing type-punned pointer will break strict-aliasing
+    rules [-Wstrict-aliasing]"
+    Besides being forbidden by rule, this cast is not needed, as register is
+    16 bits wide. Modification has been tested on L0 SoC.
+    stm32yyxx_ll_spi.h being included in soc.h file, warning is generated
+    at each compiled object, this commit allows a clean build.
+    Impacted files:
+      drivers/include/stm32l0xx_ll_spi.h
```
Comment (erwango) [inline ext/hal/st/stm32cube/stm32l0xx/README]: ST Bug tracker ID: 12974

**Review by erwango — CHANGES_REQUESTED** [review]
LGTM.
Could you update L0 README with ST bug reference I provided?

File: ext/hal/st/stm32cube/stm32l0xx/README (hunk 1 above)
Comment (tagunil) [reply ext/hal/st/stm32cube/stm32l0xx/README]: Done.

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]
Looks good.

**Review by galak — APPROVED** [review]

**Pull request merged**